free-3.md

```diff
@@ -1,50 +1,61 @@
-# Free project 3
+# Free project 3: Recursion visualizer
 
 ## The user and a language
 This section describes who the project would serve and why a language might be a
 good way to meet their needs.
 
+This project would serve intro CS students trying to wrap their heads around recursion, or more advanced students dealing with more complicated recursive algorithms. 
 
 ### What's the need?
 _What need is met by your idea? Who are you helping? What is that person's
 experience like now? What would their experience be like if you could help 
 them?_
 
+In Intro classes, recursion can be tough to grasp at first, so this would be geared towards those students. This would also benefit students in more advanced courses like Algorithms, where multiple recursive calls can be easily visualized instead of having to manually draw it out. In this way, it would be easy for students to actually see all the recursive calls saved by a lookup to a DP table, for example. 
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+CS students, the target audience for this project, are familiar with programming so using this DSL would not be hard. The idea is to allow any recursive function and its arguments to be displayed on a visualizer. It would show the starting call and subsequent calls, also displaying the arguments that change with each call.
```

Comment:
It seems like you don't really need a language for this, but instead you could (for example in Python) use the built-in AST module to hack the ast and provide graphical output.